synapse/config/server.py

```diff
@@ -410,6 +410,11 @@ def read_config(self, config, **kwargs):
             ["0.0.0.0", "::"],
             config_path=("federation_ip_range_blacklist",),
         )
+        # The federation_ip_range_whitelist is used to provide
+        # backwards-compatibility, if federation_ip_range_blacklist is set.
```

Comment:
please could you add a comment explaining why this is a sensible thing to do? It's not obvious at first glance why we would check `federation_ip_range_blacklist` when setting `federation_ip_range_whitelist`.

Reply:
It's because there never was an `federation_ip_range_whitelist` before, so if you're using the old `federation_ip_range_blacklist` config to ensure backwards compatibility you should *not* apply `ip_range_whitelist` against it.

Reply:
sorry, yes, having stared at #9569 I figured it out. What I'm saying is that this comment needs to say it, to save everyone who ever reads that bit of code having to do the same detective work.

Reply:
The comment has been updated. However, the explanation still feels a bit complex and confusing.